lib/wifi_user/use_case/check_user_is_sponsee.rb

```diff
@@ -0,0 +1,20 @@
+class WifiUser::UseCase::CheckUserIsSponsee
+  def initialize(allowlist_checker:)
+    @allowlist_checker = allowlist_checker
+  end
+
+  def execute(contact)
+    sponsee = DB[:userdetails].where(contact:)&.first
+
+    return false unless sponsee
+    return false unless sponsee[:sponsor]
```

Comment:
A user is sponsored if the contact field is not the same as the sponsor field. The sponsor field is always populated (https://github.com/alphagov/govwifi-user-signup-api/blob/7c45f2c3c60f91f1032b1de695f0a62e4d0df204/lib/wifi_user/repository/user.rb#L6 - when the sponsor is nil, it is the same as 'contact'